src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -996,10 +997,15 @@ public void RemoveTextContainer(SourceTextContainer textContainer)
                         throw new ArgumentException($"{nameof(textContainer)} is not a text container added to this project.");
                     }
 
-                    // TODO: clean up the file name if one was added
-
                     _sourceTextContainersToDocumentIds = _sourceTextContainersToDocumentIds.RemoveKey(textContainer);
 
+                    // if the TextContainer had a full path provided, remove it from the map.
+                    var (fullPath, _) = _documentPathsToDocumentIds.Where(kv => kv.Value == documentId).FirstOrDefault();
+                    if (fullPath != null)
```

Comment:
I think a .Key at the end of this would be a bit clearer than using the tuple. It's a struct so you don't have to worry about it.